### DISCUSSION TITLE
VTK and Boost development

### DISCUSSION THREAD

**Discussion by joeymukherjee** [Homebrew/discussions#3499] [Everyday usage]
I'm interested in installing VTK 9.1 libraries and I have Boost upgraded to the latest version (1.79).  Unfortunately, the VTK 9.1 available requires 1.78 of Boost.  

-- Could NOT find Boost: Found unsuitable version "1.79.0", but required is exact version "1.78.0" (found /usr/local/include, )
-- Could not find the VTK package due to a missing dependency: Boost
-- Could NOT find Boost: Found unsuitable version "1.79.0", but required is exact version "1.78.0" (found /usr/local/include, found components: serialization)

is the error I am getting from CMake.  

Is there an easy way to solve this issue?  Building VTK can be a challenge, but so can boost, but maybe that's easier?  I'm on Big Sur if that matters.

**Comment by SMillerDev** [comment]
Have you tried asking the vtk developers to cut a release compatible with the latest boost version?

**Comment by gromgit** [comment]
> I'm interested in installing VTK 9.1 libraries and I have Boost upgraded to the latest version (1.79). Unfortunately, the VTK 9.1 available requires 1.78 of Boost.

Assuming you're using the Homebrew versions of both `vtk` and `boost`, that indeed seems to be the problem:
```
$ rg -C3 -F 1.78 /opt/homebrew/opt/vtk/
/opt/homebrew/opt/vtk/lib/cmake/vtk-9.1/VTK-vtk-module-find-packages.cmake
1006-  endif ()
1007-
1008-  find_package(Boost
1009:    1.78.0
1010-    EXACT
1011-    
1012-    ${_vtk_module_find_package_quiet}
--
1053-  endif ()
1054-
1055-  find_package(Boost
1056:    1.78.0
1057-    EXACT
1058-    
1059-    ${_vtk_module_find_package_quiet}
```
Normally, `vtk` should've been revision-bumped along with the new `boost`, but no one caught it likely because its `test` block _doesn't involve Boost at all_.

**Reply by carlocab** [reply]
Thanks for finding that, @gromgit.

Should hopefully be fixed by https://github.com/Homebrew/homebrew-core/pull/107456.

**Comment by carlocab** [comment]
This should be fixed now: `brew update && brew upgrade vtk`.
